### PR TITLE
click_events

### PR DESCRIPTION
**Pull request by akokshar**
Implementing click_events feature. Followed documentation at https://i3wm.org/docs/i3bar-protocol.html
Right, Left, Middle clicks on a status block trigger an event to be send to the standard input of status_command

### CODE REVIEW

File: swaybar/bar.c [hunk 1]
```diff
@@ -31,16 +32,30 @@ static void bar_init(struct bar *bar) {
...
+		int pipe_write_fd[2];
+
+		if (pipe(pipe_read_fd) != 0) {
+			sway_log(L_ERROR, "Unable to create pipes for status_command fork");
+			return;
+		}
+		if (pipe(pipe_write_fd) != 0) {
+			sway_log(L_ERROR, "Unable to create pipe for status_command fork (write)");
+			close(pipe_read_fd[0]);
+			close(pipe_read_fd[1]);
```
Comment (SirCmpwn) [inline swaybar/bar.c]: These changes seem unrelated?

File: swaybar/status_line.c [hunk 2]
```diff
@@ -391,6 +393,67 @@ static int i3json_handle_fd(struct bar *bar) {
...
+
+		// find clicked block
+		struct status_block *clicked_block = NULL;
+		struct status_block *current_block = NULL;
+		int num_blocks = bar->status->block_line->length;
+
+		if (num_blocks == 0) {
+			return false;
+		}
+		else {
```
Comment (SirCmpwn) [inline swaybar/status_line.c]: Style errors, review the style guide

**Comment by SirCmpwn** [comment]
LGTM, aside from a few style issues and those unrelated changes. Can you pull them into a separate PR?

File: swaybar/bar.c (hunk 1 above)
Comment (akokshar) [reply swaybar/bar.c]: These are related. 
There were no pipe created for the stdin of a spawned process. The lines marked above are part of a bigger change which defines pipe_write_fd and attaches if to stdin of a status command

Comment (SirCmpwn) [reply swaybar/bar.c]: Ah, fair enough.

File: swaybar/render.c [hunk 3]
```diff
@@ -94,6 +94,9 @@ static void render_block(struct window *window, struct config *config, struct st
 
 	double pos = *x;
 
+	block->x = (int) pos;
+	block->width = (int) block_width;
```
Comment (SirCmpwn) [inline swaybar/render.c]: That wasn't the only style issue. Check here as well.

File: swaybar/status_line.c [hunk 4]
```diff
@@ -418,15 +480,29 @@ bool handle_status_line(struct bar *bar) {
...
 				) {
 					sway_log(L_DEBUG, "Switched to i3bar protocol.");
 					bar->status->protocol = I3BAR;
-					i3json_handle_data(bar, line_rest);
 				}
+
+				json_object *click_events;
+				if (json_object_object_get_ex(proto, "click_events", &click_events)
+							&& json_object_get_boolean(click_events)
+				) {
```
Comment (SirCmpwn) [inline swaybar/status_line.c]: Here as well

Comment (SirCmpwn) [reply swaybar/status_line.c]: No, no, put the parens and brace on the same line.

Comment (akokshar) [reply swaybar/status_line.c]: Ah, ok. 
I have just made same thing as in a code right above. Condition there is a multi-line one. 
if I do it on a same line it is 116 chars long...
Can I split it like this:
if (json_object_object_get_ex(proto, "click_events", &click_events) 
      && json_object_get_boolean(click_events)) {

or just keep it 116 chars long?

Comment (SirCmpwn) [reply swaybar/status_line.c]: I would split it on the boolean operator and let it exceed 80 columns if necessary.

**Comment by akokshar** [comment]
@SirCmpwn I am done with a changes you requested. Could you review again please.

**Comment by SirCmpwn** [comment]
Thanks!
